packages/dx-react-scheduler/src/plugins/drag-drop-provider.jsx

```diff
@@ -226,26 +178,30 @@ export class DragDropProvider extends React.PureComponent {
             }, {
               commitChangedAppointment, changeAppointment,
               startEditAppointment, stopEditAppointment,
-            }) => (
-              <DragDropProviderCore
-                onChange={this.onPayloadChange({ commitChangedAppointment, stopEditAppointment })}
-              >
-                <DropTarget
-                  onOver={this.onOver({
-                    viewCellsData,
-                    startViewDate,
-                    endViewDate,
-                    excludedDays,
-                    timeTableElement,
-                    layoutElement,
-                    layoutHeaderElement,
-                  }, { changeAppointment, startEditAppointment, stopEditAppointment })}
-                  onDrop={this.onDrop({ commitChangedAppointment, stopEditAppointment })}
+            }) => {
+              const calculateBoundariesByMove = this.calculateNextBoundaries({
+                viewCellsData,
+                startViewDate,
+                endViewDate,
+                excludedDays,
+                timeTableElement,
+                layoutElement,
+                layoutHeaderElement,
+              }, { changeAppointment, startEditAppointment, stopEditAppointment });
+              return (
+                <DragDropProviderCore
+                  onChange={this.onPayloadChange({ commitChangedAppointment, stopEditAppointment })}
                 >
-                  <TemplatePlaceholder />
-                </DropTarget>
-              </DragDropProviderCore>
-            )}
+                  <DropTarget
+                    onOver={calculateBoundariesByMove}
+                    onEnter={calculateBoundariesByMove}
+                    onDrop={this.onDrop({ commitChangedAppointment, stopEditAppointment })}
```

Comment:
Can we call `handleDrop` directly? `onDrop` just returns a wrapper around `handleDrop`.

Reply:
No, the `onDrop` property required a function and call it with argument `{ payload, clientOffset }`. In this case, we should pass our actions to this function